### PR TITLE
Replace binding type for ease of use in TypeScript

### LINKED ISSUE #5120: Replace UInt8Array by strings in bindings for EntryID, RealmID and InvitationToken
Strings are easier to manipulate in typescript, and it's easier to serialize/deserialize those byte arrays in Rust that it is in Typescript, so everything that is a Uint8Array in the binding (RealmID, EntryID and InvitationToken) should be changed to string.

https://github.com/Scille/parsec-cloud/blob/556e3328a9ca1ea49559e0cde9b4afbfeb7cccbc/bindings/generator/api/common.py#L166-L181
Should inherit `StringBased` instead of `BytesBased` and change the convert function accordingly.

Preferably, the string should be human-readable (hash-like or base64 or something like that) as it will be used in URLs for routing (`/workspaces/<WorkspaceID>`) for example.

### PR DESCRIPTION
**Pull request by mmmarcos**
## Describe your changes

* `EntryID(BytesBasedType)` --> `EntryID(StrBasedType)`
* `RealmID(BytesBasedType)` --> `RealmID(StrBasedType)`
* `InvitationToken(BytesBasedType)` --> `InvitationToken(StrBasedType)`

Closes #5120

## Checklist

Before you submit this pull request, please make sure to:

- [x] Keep changes in the pull request as small as possible
- [x] Ensure the commit history is sanitized
- [x] Give a meaningful title to your PR
- [x] Describe your changes
- [x] Link any related issue in the description

### CODE REVIEW

File: bindings/generator/api/common.py [hunk 1]
```diff
@@ -163,22 +163,14 @@ class EntryName(StrBasedType):
...
-class EntryID(BytesBasedType):
-    custom_from_rs_bytes = (
-        "|x: &[u8]| -> Result<_, _> { libparsec::EntryID::try_from(x).map_err(|e| e.to_string()) }"
-    )
-    custom_to_rs_bytes = (
-        "|x: libparsec::EntryID| -> Result<_, &'static str> { Ok(x.as_bytes().to_owned()) }"
-    )
+class EntryID(StrBasedType):
+    custom_from_rs_bytes = "|x: &[u8]| -> Result<_, _> { libparsec::EntryID::try_from(x).map_err(|e| e.to_string()).as_hyphenated() }"
+    custom_to_rs_bytes = "|s: String| -> Result<_, &'static str> { Ok(s.as_bytes().to_owned()) }"
```
Comment (FirelightFlagboy) [inline bindings/generator/api/common.py:168]:
```suggestion
    custom_to_rs_bytes = "|s: String| -> Result<libparsec::EntryID, &'static str> { Ok(s.as_bytes().to_owned()) }"
```

File: bindings/generator/api/common.py [hunk 2]
```diff
@@ -163,22 +163,14 @@ class EntryName(StrBasedType):
...
 
-class RealmID(BytesBasedType):
-    custom_from_rs_bytes = (
-        "|x: &[u8]| -> Result<_, _> { libparsec::RealmID::try_from(x).map_err(|e| e.to_string()) }"
-    )
-    custom_to_rs_bytes = (
-        "|x: libparsec::RealmID| -> Result<_, &'static str> { Ok(x.as_bytes().to_owned()) }"
-    )
+class RealmID(StrBasedType):
+    custom_from_rs_bytes = "|x: &[u8]| -> Result<_, _> { libparsec::RealmID::try_from(x).map_err(|e| e.to_string()).as_hyphenated() }"
```
Comment (FirelightFlagboy) [inline bindings/generator/api/common.py:172]:
```suggestion
    custom_from_rs_bytes = "|x: &[u8]| -> Result<String, _> { libparsec::RealmID::try_from(x).map_err(|e| e.to_string()).as_hyphenated() }"
```

File: bindings/generator/api/common.py [hunk 3]
```diff
@@ -163,21 +163,22 @@ class EntryName(StrBasedType):
...
-    custom_from_rs_bytes = (
-        "|x: &[u8]| -> Result<_, _> { libparsec::EntryID::try_from(x).map_err(|e| e.to_string()) }"
-    )
+# EntryID and RealmID are defined as strings (instead of Uint8Array) so that the
+# Typescript code only manipulates strings without conversion or parsing. Also,
+# they are used for routing (e.g /workspace/<id>) so a Simple format is used.
+
+
+class EntryID(StrBasedType):
+    custom_from_rs_bytes = "|x: &[u8]| -> Result<String, _> { libparsec::EntryID::try_from(x).map_err(|e| e.to_string()).hex() }"
```
Comment (FirelightFlagboy) [inline bindings/generator/api/common.py:172]:
```suggestion
    custom_from_rs_str = "|x: &str| -> Result<String, _> { libparsec::EntryID::try_from(x).map_err(|e| e.to_string()).hex() }"
```

File: bindings/generator/api/common.py [hunk 4]
```diff
@@ -163,21 +163,22 @@ class EntryName(StrBasedType):
...
+        "|s: String| -> Result<libparsec::EntryID, &'static str> { Ok(s.as_bytes().to_owned()) }"
     )
 
 
-class RealmID(BytesBasedType):
-    custom_from_rs_bytes = (
-        "|x: &[u8]| -> Result<_, _> { libparsec::RealmID::try_from(x).map_err(|e| e.to_string()) }"
-    )
+class RealmID(StrBasedType):
+    custom_from_rs_bytes = "|x: &[u8]| -> Result<String, _> { libparsec::RealmID::try_from(x).map_err(|e| e.to_string()).hex() }"
```
Comment (FirelightFlagboy) [inline bindings/generator/api/common.py:179]:
```suggestion
    custom_from_rs_str = "|x: &str| -> Result<String, _> { libparsec::RealmID::try_from(x).map_err(|e| e.to_string()).hex() }"
```

File: bindings/generator/api/common.py [hunk 5]
```diff
@@ -163,21 +163,22 @@ class EntryName(StrBasedType):
...
+# Typescript code only manipulates strings without conversion or parsing. Also,
+# they are used for routing (e.g /workspace/<id>) so a Simple format is used.
+
+
+class EntryID(StrBasedType):
+    custom_from_rs_bytes = "|x: &[u8]| -> Result<String, _> { libparsec::EntryID::try_from(x).map_err(|e| e.to_string()).hex() }"
     custom_to_rs_bytes = (
-        "|x: libparsec::EntryID| -> Result<_, &'static str> { Ok(x.as_bytes().to_owned()) }"
+        "|s: String| -> Result<libparsec::EntryID, &'static str> { Ok(s.as_bytes().to_owned()) }"
     )
```
Comment (FirelightFlagboy) [inline bindings/generator/api/common.py:175]:
```suggestion
    custom_to_rs_string = (
         "|x: libparsec::EntryID| -> Result<String, &'static str> { Ok(x.hex()) }
    )
```

File: bindings/generator/api/common.py [hunk 6]
```diff
@@ -163,21 +163,22 @@ class EntryName(StrBasedType):
...
-class RealmID(BytesBasedType):
-    custom_from_rs_bytes = (
-        "|x: &[u8]| -> Result<_, _> { libparsec::RealmID::try_from(x).map_err(|e| e.to_string()) }"
-    )
+class RealmID(StrBasedType):
+    custom_from_rs_bytes = "|x: &[u8]| -> Result<String, _> { libparsec::RealmID::try_from(x).map_err(|e| e.to_string()).hex() }"
     custom_to_rs_bytes = (
-        "|x: libparsec::RealmID| -> Result<_, &'static str> { Ok(x.as_bytes().to_owned()) }"
+        "|s: String| -> Result<libparsec::RealmID, &'static str> { Ok(s.as_bytes().to_owned()) }"
     )
```
Comment (FirelightFlagboy) [inline bindings/generator/api/common.py:182]:
```suggestion
    custom_to_rs_string = (
        "|x: libparsec::RealmID| -> Result<String, &'static str> { Ok(x.hex()) }
    )
```

**Review by FirelightFlagboy — APPROVED** [review]

**Review by Max-7 — APPROVED** [review]
Only tested with `RealmID` for now but it should be alright for the two other types.

**Pull request merged**